src/simplnx/Utilities/GeometryHelpers.cpp

```diff
@@ -24,5 +23,152 @@ std::string GenerateGeometryInfo(const nx::core::SizeVec3& dims, const nx::core:
+
+namespace nx::core::GeometryHelpers::Connectivity
+{
+std::vector<int32> FindEulerCharacteristicValues(const TriangleGeom& triangleGeom, const Int32Array& faceLabelsRef)
+{
+  const auto& triangleList = triangleGeom.getFacesRef().getDataStoreRef();
+  const auto& faceLabels = faceLabelsRef.getDataStoreRef();
+  const usize numRegions = 1 + *std::max_element(faceLabels.begin(), faceLabels.end());
+
+  using EdgePairType = std::pair<IGeometry::MeshIndexType, IGeometry::MeshIndexType>;
+  using UniqueEdgesType = std::set<EdgePairType>;
+  using UniqueVertType = std::set<IGeometry::MeshIndexType>;
+
+  constexpr IGeometry::MeshIndexType numVertsPerElem = 3;
+  std::vector<int64> regionTriangleCount(numRegions, 0);
+  std::vector<UniqueEdgesType> uniqueEdges(numRegions);
+  std::vector<UniqueVertType> uniqueVerts(numRegions);
+
+  for(IGeometry::MeshIndexType tIdx = 0; tIdx < triangleList.getNumberOfTuples(); ++tIdx)
+  {
+    const usize offset = tIdx * numVertsPerElem;
+
+    for(IGeometry::MeshIndexType labelIdx = 0; labelIdx < 2; labelIdx++)
+    {
+      IGeometry::MeshIndexType v0 = 0;
+      IGeometry::MeshIndexType v1 = 0;
+
+      auto regionIdx = faceLabels[tIdx * 2 + labelIdx];
+      if(regionIdx < 0)
+      {
+        continue;
+      }
+
+      ++regionTriangleCount[regionIdx];
+
+      for(usize j = 0; j < numVertsPerElem; j++)
+      {
+        if(j == (numVertsPerElem - 1))
+        {
+          if(triangleList[offset + j] > triangleList[offset + 0])
+          {
+            v0 = triangleList[offset + 0];
+            v1 = triangleList[offset + j];
+          }
+          else
+          {
+            v0 = triangleList[offset + j];
+            v1 = triangleList[offset + 0];
+          }
+        }
+        else
+        {
+          if(triangleList[offset + j] > triangleList[offset + j + 1])
+          {
+            v0 = triangleList[offset + j + 1];
+            v1 = triangleList[offset + j];
+          }
+          else
+          {
+            v0 = triangleList[offset + j];
+            v1 = triangleList[offset + j + 1];
+          }
+        }
+        EdgePairType edge = std::make_pair(v0, v1);
+        uniqueVerts[regionIdx].insert(v0);
+        uniqueVerts[regionIdx].insert(v1);
+        uniqueEdges[regionIdx].insert(edge);
+      }
+    }
+  } // End triangle Loop
+
+  std::vector<int32> eulerCharacteristicValues(regionTriangleCount.size(), 0);
+  for(usize i = 0; i < regionTriangleCount.size(); i++)
+  {
+    eulerCharacteristicValues[i] = static_cast<int32>(static_cast<int64>(uniqueVerts[i].size()) + regionTriangleCount[i] - static_cast<int64>(uniqueEdges[i].size()));
+  }
+  //    std::string message = fmt::format("Region: {} Euler Characteristic: {} = V:{} + F:{} - E:{}", regionIdx, eulerCharacteristic, uniqueVerts.size(), numTriangles, uniqueEdges.size());
+  //    m_MessageHandler(IFilter::Message::Type::Info, message);
+  return eulerCharacteristicValues;
+}
+
+#if 0
```

Comment:
If this is being preserved for historical context or future integration, a comment giving pretense should be added, else remove.